lib/service/connect.go

```diff
@@ -1066,8 +1064,17 @@ func (process *TeleportProcess) newClient(authServers []utils.NetAddr, identity
 	}
 	tunnelClient, err := process.newClientThroughTunnel(authServers, tlsConfig, sshClientConfig)
 	if err != nil {
-		directErrLogger.Debug("Failed to connect to Auth Server directly.")
-		logger.WithError(err).Debug("Failed to connect to Auth Server through tunnel.")
+		process.log.Errorf("Node failed to establish connection to Teleport Proxy. We have tried the following endpoints:")
+		// Can't errors.As directErr in the "x509: certificate is valid for x but not y" error case, as only message field is set
+		if trace.IsConnectionProblem(directErr) && strings.Contains(directErr.Error(), "x509: certificate is valid for") {
+			directErr = trace.Wrap(directErr, "TLS certificate error. Certificate is invalid or not trusted. "+
+				"Fix the certificate to correct this error: https://goteleport.com/docs/architecture/authentication/")
+		}
+		process.log.Errorf("- connecting to auth server directly: %v", directErr)
+		if trace.IsConnectionProblem(err) && strings.Contains(err.Error(), "connection refused") {
+			err = trace.Wrap(err, "This is the alternative port we tried and it's not configured.")
```

Comment:
What's "this" referring to? Should we mention some port number that was tried to help debug, or will it be pretty evident in context?

Reply:
The port is included in the connection refused error that is being wrapped here.
Unless there is a case I am unaware of and was not able to find it should be pretty clear to the user from the context.